src/credentials/auth.ts

```diff
@@ -760,16 +762,17 @@ export class Auth implements AuthService, ConnectionManager {
         )
     }
 
-    private getIamConnection(id: Connection['id'], provider: CredentialsProvider): IamConnection & StatefulConnection {
-        const profile = this.store.getProfileOrThrow(id)
-
+    private getIamConnection(
```

Comment:
is it worth adding a comment here to call out the perf implications?

```suggestion
    /** Reads the filesystem on each call. */
    private getIamConnection(
```

Reply:
So this specific method doesn't make any `fs` calls, it's only when you do `getCredentials` on shared ini credentials. I think callers should be mindful of _any_ `async` operation as potentially impacting perf.

In general, VS Code API designs avoid making `get` operations asynchronous. Which is something we should strive for as well. Designing these types of systems can be a lot of effort though because we have to be aware of all operations, including those not within our program, that affect the state of the system.